File: notifications/src/main/java/io/castled/notifications/trigger/models/PropertyOperation.java
package io.castled.notifications.trigger.models;

import io.castled.notifications.trigger.enums.OperationType;
import io.castled.notifications.trigger.enums.PropertyType;

public class PropertyOperation {

    private final OperationType operationType;
    private final PropertyType propertyType;

    public PropertyOperation(OperationType operationType, PropertyType propertyType) {
        this.operationType = operationType;
        this.propertyType = propertyType;
    }

    public OperationType getOperationType() {
        return operationType;
    }

    public PropertyType getPropertyType() {
        return propertyType;
    }

}

File: notifications/src/main/java/io/castled/notifications/trigger/models/PropertyOperation.kt
package io.castled.notifications.trigger.models

import io.castled.notifications.trigger.enums.OperationType
import io.castled.notifications.trigger.enums.PropertyType
import kotlinx.serialization.Polymorphic
import kotlinx.serialization.Serializable

@Serializable
@Polymorphic
internal sealed class PropertyOperation(val operationType: OperationType, val propertyType: PropertyType)